mithril-common/src/crypto_helper/cardano/opcert.rs

```diff
@@ -0,0 +1,76 @@
+use super::FromShelleyFile;
+
+use ed25519_dalek::{PublicKey as EdPublicKey, Signature as EdSignature, Verifier};
+use kes_summed_ed25519::common::PublicKey as KesPublicKey;
+use mithril::RegisterError;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+/// Raw Fields of the operational certificates (without incluiding the cold VK)
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+struct RawFields(
+    #[serde(with = "serde_bytes")] Vec<u8>,
+    u64,
+    u64,
+    #[serde(with = "serde_bytes")] Vec<u8>,
+);
+
+/// Raw Operational Certificate
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+struct RawOpCert(RawFields, EdPublicKey);
+
+impl FromShelleyFile for RawOpCert {}
+
+/// Parsed Operational Certificate
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct OpCert {
+    pub(crate) kes_vk: KesPublicKey,
+    pub(crate) issue_number: u64,
+    pub(crate) start_kes_period: u64, // this is not the kes period used in signing/verifying
+    pub(crate) cert_sig: EdSignature,
+    pub(crate) cold_vk: EdPublicKey,
+}
+
+impl OpCert {
```

Comment:
@iquerejeta the`OpCert` needs a `to_file` function as well, as agreed during pairing